projects/igniteui-angular/src/lib/directives/focus-trap/focus-trap.directive.ts

```diff
@@ -0,0 +1,88 @@
+import { AfterViewInit, Directive, ElementRef, NgModule, OnDestroy } from '@angular/core';
+import { fromEvent, Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import { PlatformUtil } from '../../core/utils';
+
+@Directive({
+    selector: '[igxFocusTrap]'
+})
+export class IgxFocusTrapDirective implements AfterViewInit, OnDestroy {
+     /** @hidden */
+    public get element(): HTMLElement | null {
+        return this.elementRef.nativeElement;
+    }
+
+    private destroy$ = new Subject();
+
+    /** @hidden */
+    constructor(
+        private elementRef: ElementRef,
+        protected platformUtil: PlatformUtil) {
+    }
+
+    /** @hidden */
+    public ngAfterViewInit(): void {
+        fromEvent(this.element, 'keydown')
+            .pipe(takeUntil(this.destroy$))
+            .subscribe((event: KeyboardEvent) => {
+                if (event.key === this.platformUtil.KEYMAP.TAB) {
+                    this.handleTab(event);
+                }
+            });
+    }
+
+    /** @hidden */
+    public ngOnDestroy() {
+        this.destroy$.next();
+        this.destroy$.complete();
+    }
+
+    private handleTab(event) {
+        const elements = this.getFocusableElements(this.element);
+        if (elements.length > 0) {
```

Comment:
in an edge case when there are no focusable elements and one still needs to trap focus we should focus `this.element`. So add else clause here and focus `this.element` there.